trie/triedb/hashdb/database.go

```diff
@@ -633,97 +595,38 @@ func (db *Database) Update(nodes *MergedNodeSet) error {
 
 // Size returns the current storage size of the memory cache in front of the
 // persistent database layer.
-func (db *Database) Size() (common.StorageSize, common.StorageSize) {
+func (db *Database) Size() common.StorageSize {
 	db.lock.RLock()
 	defer db.lock.RUnlock()
 
 	// db.dirtiesSize only contains the useful data in the cache, but when reporting
 	// the total memory consumption, the maintenance metadata is also needed to be
 	// counted.
 	var metadataSize = common.StorageSize(len(db.dirties) * cachedNodeSize)
-	var preimageSize common.StorageSize
-	if db.preimages != nil {
-		preimageSize = db.preimages.size()
-	}
-	return db.dirtiesSize + db.childrenSize + metadataSize, preimageSize
+	return db.dirtiesSize + db.childrenSize + metadataSize
 }
 
-// GetReader retrieves a node reader belonging to the given state root.
-func (db *Database) GetReader(root common.Hash) Reader {
-	return newHashReader(db)
+// Close closes the trie database and releases all held resources.
+func (db *Database) Close() error { return nil }
```

Comment:
Hmm, does the path based trie need a close method?

Reply:
Yes. Pathdb will retain a freezer and needs to be closed when db is closed.